File: injector/src/test/java/net/mready/inject/TransitiveDependencyTest.java
package net.mready.inject;

import org.junit.Test;

import javax.inject.Inject;

import static org.junit.Assert.assertNotNull;

public class TransitiveDependencyTest {
    @Test
    public void transitive() {
        Injector injector = Injector.with();
        A a = injector.instance(A.class);
        assertNotNull(a.b.c);
    }

    public static class A {
        private final B b;

        @Inject
        public A(B b) {
            this.b = b;
        }
    }

    public static class B {
        private final C c;

        @Inject
        public B(C c) {
            this.c = c;
        }
    }

    public static class C {

    }


}

File: injector/src/test/kotlin/net/mready/inject/TransitiveDependencyTest.kt
package net.mready.inject

import org.junit.Assert.assertNotNull
import org.junit.Test

class TransitiveDependencyTest {
    @Test
    fun transitive() {
        val injector = Injector.with()
        val a = injector.instance(A::class.java)
        assertNotNull(a.b.c)
    }

    class A(val b: B)

    class B(val c: C)

    class C

}
